arc/matrices/tests/test_csr.py

```diff
@@ -1,140 +1,231 @@
 import pytest
 from arc.matrices import csr_matrix
+from arc.matrices import data_types
 
 
 def test_default_constructor():
-    test_input_matrix = csr_matrix(3, 3)
-    assert test_input_matrix.rows == 3
-    assert test_input_matrix.cols == 3
-    expected_matrix = [[0] * 3 for _ in range(3)]
+    try:
+        matrix_instance = csr_matrix(3, 3)
 
-    assert test_input_matrix.matrix == expected_matrix
+        assert matrix_instance.rows == 3, f"Expected 3 rows, but got {matrix_instance.rows}"
+        assert matrix_instance.cols == 3, f"Expected 3 columns, but got {matrix_instance.cols}"
 
+        expected_matrix = [[0] * 3 for _ in range(3)]
+        assert matrix_instance.matrix == expected_matrix, f"Expected {expected_matrix}, but got {matrix_instance.matrix}"
+
+        assert isinstance(matrix_instance, csr_matrix), "The object is not an instance of csr_matrix"
+
+    except Exception as e:
+        assert False, f"An error occurred: {e}"
 
 def test_all():
-    test_input_matrix = csr_matrix(3, 3, data=[1, 2, 3, 4, 5, 6], row=[
-                                   0, 0, 1, 2, 2, 2], col=[0, 2, 2, 0, 1, 2])
-    expected_matrix = [[1, 0, 2], [0, 0, 3], [4, 5, 6]]
+    try:
+        matrix_instance = csr_matrix(3, 3, data=[1, 2, 3, 4, 5, 6], row=[0, 0, 1, 2, 2, 2], col=[0, 2, 2, 0, 1, 2])
+        expected_matrix = [[1, 0, 2], [0, 0, 3], [4, 5, 6]]
 
-    assert test_input_matrix.matrix == expected_matrix
+        assert matrix_instance.matrix == expected_matrix, f"Expected {expected_matrix}, but got {matrix_instance.matrix}"
 
+        assert isinstance(matrix_instance, csr_matrix), "The object is not an instance of csr_matrix"
 
-def test_get_cols():
-    test_input_matrix = csr_matrix(3, 3).get_cols()
+    except Exception as e:
+        assert False, f"An error occurred: {e}"
+
+def test_data():
```

Comment:
I originally removed this test case due to: 
```python3 
if len(data) != rows * cols:
                    raise ValueError("Length of data must match rows * cols.")
``` 

Wouldn't this still throw the ValueError?